apps/examples/wifi_manager_sample/wm_test.c

```diff
@@ -425,140 +317,99 @@ static wifi_manager_ap_crypto_type_e get_crypto_type(const char *method)
+void _wt_sta_connected(wifi_manager_result_e res)
 {
-	printf("[WT]  T%d --> %s res(%d)\n", getpid(), __FUNCTION__, res);
-	WM_TEST_SIGNAL;
+	WT_LOG(TAG, "-->res(%d)", res);
+	WT_TEST_SIGNAL;
 }
 
-void wm_sta_disconnected(wifi_manager_disconnect_e disconn)
+void _wt_sta_disconnected(wifi_manager_disconnect_e disconn)
 {
-	sleep(2);
-	printf("[WT]  T%d --> %s\n", getpid(), __FUNCTION__);
-	WM_TEST_SIGNAL;
+	WT_LOG(TAG, "-->");
+	WT_TEST_SIGNAL;
 }
 
-void wm_softap_sta_join(void)
+void _wt_softap_sta_join(void)
 {
-	printf("[WT]  T%d --> %s\n", getpid(), __FUNCTION__);
-	WM_TEST_SIGNAL;
+	WT_LOG(TAG, "-->");
+	WT_TEST_SIGNAL;
 }
 
-void wm_softap_sta_leave(void)
+void _wt_softap_sta_leave(void)
 {
-	printf("[WT]  T%d --> %s\n", getpid(), __FUNCTION__);
-	WM_TEST_SIGNAL;
+	WT_LOG(TAG, "-->");
+	WT_TEST_SIGNAL;
 }
 
-void wm_scan_done(wifi_manager_scan_info_s **scan_result, wifi_manager_scan_result_e res)
+void _wt_scan_done(wifi_manager_scan_info_s **scan_result, wifi_manager_scan_result_e res)
 {
-	printf("[WT]  T%d --> %s\n", getpid(), __FUNCTION__);
+	WT_LOG(TAG, "-->");
 	/* Make sure you copy the scan results onto a local data structure.
 	 * It will be deleted soon eventually as you exit this function.
 	 */
 	if (scan_result == NULL) {
-		WM_TEST_SIGNAL;
+		WT_TEST_SIGNAL;
 		return;
 	}
-	wifi_manager_scan_info_s *wifi_scan_iter = *scan_result;
-	while (wifi_scan_iter != NULL) {
-		printf("[WT] WiFi AP SSID: %-25s, BSSID: %-20s, Rssi: %d, Auth: %s, Crypto: %s\n",
-			   wifi_scan_iter->ssid, wifi_scan_iter->bssid, wifi_scan_iter->rssi,
-			   wifi_test_auth_method[wifi_scan_iter->ap_auth_type],
-			   wifi_test_crypto_method[wifi_scan_iter->ap_crypto_type]);
-		wifi_scan_iter = wifi_scan_iter->next;
-	}
-	WM_TEST_SIGNAL;
+	_wt_print_scanlist(*scan_result);
+	WT_TEST_SIGNAL;
 }
 
-/*
- * Control Functions
- */
-void wm_start(void *arg)
+/* Control Functions */
+void _wt_start(void *arg)
 {
-	WM_TEST_LOG_START;
+	WT_ENTER;
 	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
 
-	res = wifi_manager_init(&wifi_callbacks);
+	res = wifi_manager_init(&g_wifi_callbacks);
 	if (res != WIFI_MANAGER_SUCCESS) {
-		printf("[WT]  wifi_manager_init fail\n");
+		WT_LOGE(TAG, "wifi_manager_init fail");
 	}
-	WM_TEST_LOG_END;
+	WT_LEAVE;
 }
 
-void wm_stop(void *arg)
+void _wt_stop(void *arg)
 {
-	WM_TEST_LOG_START;
+	WT_ENTER;
 	wifi_manager_result_e res = wifi_manager_deinit();
 	if (res != WIFI_MANAGER_SUCCESS) {
-		printf("[WT]  WiFi Manager failed to stop\n");
+		WT_LOGE(TAG, "WiFi Manager failed to stop");
 	}
-	WM_TEST_LOG_END;
+	WT_LEAVE;
 }
 
-void wm_softap_start(void *arg)
+void _wt_softap_start(void *arg)
 {
-	WM_TEST_LOG_START;
+	WT_ENTER;
 	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
-	struct options *ap_info = (struct options *)arg;
+	struct wt_options *ap_info = (struct wt_options *)arg;
 	if (strlen(ap_info->ssid) > WIFIMGR_SSID_LEN || strlen(ap_info->password) > WIFIMGR_PASSPHRASE_LEN) {
```

Comment:
I think `ap_info` can be dereferencing.

Reply:
I can't understand your point? _wt_process pass wt_options here.

Reply:
I mean, `ap_info` can be NULL, then `ap_info->xxx' will make assert.

Reply:
if parsing were fail then _wt_softap_start wouldn't run. So it's safe.